<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="fixed_base" default="true"/>
    <arg name="object"  default="door"/>
    <arg name="description_file" default="$(find mppi_manipulation)/data/panda_grasp.urdf.xacro"/>
    <arg name="tf_prefix" default=""/>

    <param name="robot_description"
           command="$(find xacro)/xacro $(arg description_file) resources_prefix:='file://' add_finger:=true fixed_base:=$(arg fixed_base)"/>
    <param name="robot_description_raisim" command="$(find xacro)/xacro $(arg description_file)
      collision_links:=true
      collision_hand:=true
      collision_finger:=true
      collision_format:='obj'
      add_finger:=true
      no_position_limits:=true
      fixed_base:=$(arg fixed_base)"/>

    <param name="object_description" command="$(find xacro)/xacro $(find mppi_manipulation)/data/$(arg object).urdf.xacro  resource_prefix:='file://'"/>
    <param name="object_description_raisim"
           command="$(find xacro)/xacro $(find mppi_manipulation)/data/$(arg object).urdf.xacro"/>


    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="100"/>
        <param name="use_tf_static" value="true"/>
        <param name="tf_prefix" value="$(arg tf_prefix)"/>
        <remap from="robot_description" to="robot_description"/>
        <remap from="joint_states" to="/joint_states"/>
    </node>

    <node name="object_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="use_tf_static" value="true"/>
        <param name="publish_frequency" value="100"/>
        <remap from="robot_description" to="object_description"/>
        <remap from="joint_states" to="/object/joint_state"/>
    </node>

    <node name="manipulation_raisim" type="manipulation_raisim" pkg="mppi_manipulation" output="screen"
          launch-prefix="gdb -ex run --args">
        <param name="activation_file" value="/home/giuseppe/raisim_ws/raisimLib/build/examples/rsc/activation.raisim"/>
        <param name="fixed_base"    value="$(arg fixed_base)"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mppi_manipulation)/config/config.rviz" output="screen"/>

    <node name="rqt_multiplot" pkg="rqt_multiplot" type="rqt_multiplot"
          args="--multiplot-config $(find mppi_manipulation)/config/joint_velocity_tracking.xml"
          output="screen"/>

</launch>
